File: docs-examples/example-java/src/test/java/io/micronaut/jms/docs/AbstractJmsSpec.java
package io.micronaut.jms.docs;

import io.micronaut.context.ApplicationContext;
import io.micronaut.inject.qualifiers.Qualifiers;
import io.micronaut.jms.pool.JMSConnectionPool;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

import static io.micronaut.jms.activemq.classic.configuration.ActiveMqClassicConfiguration.CONNECTION_FACTORY_BEAN_NAME;

public abstract class AbstractJmsSpec {

    protected ApplicationContext applicationContext;

    @BeforeEach
    void setup() {
        applicationContext = ApplicationContext.run(getConfiguration(), "test");
    }

    @AfterEach
    void cleanup() {
        if (applicationContext != null) {
            applicationContext.close();
        }
    }

    protected Map<String, Object> getConfiguration() {
        String broker = UUID.randomUUID().toString().replaceAll("-", "");

        Map<String, Object> config = new HashMap<>();
        config.put("micronaut.jms.activemq.classic.connectionString", "vm://" + broker + "?broker.persistent=false");
        config.put("micronaut.jms.activemq.classic.enabled", true);
        config.put("spec.name", getClass().getSimpleName());
        return config;
    }

    protected JMSConnectionPool getConnectionPool() {
        return applicationContext.getBean(JMSConnectionPool.class, Qualifiers.byName(CONNECTION_FACTORY_BEAN_NAME));
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/jms/docs/AbstractJmsSpec.kt
package io.micronaut.jms.docs

import io.micronaut.context.ApplicationContext
import io.micronaut.inject.qualifiers.Qualifiers
import io.micronaut.jms.activemq.classic.configuration.ActiveMqClassicConfiguration.CONNECTION_FACTORY_BEAN_NAME
import io.micronaut.jms.pool.JMSConnectionPool
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import java.util.UUID

abstract class AbstractJmsSpec {

    protected lateinit var applicationContext: ApplicationContext

    @BeforeEach
    fun setup() {
        applicationContext = ApplicationContext.run(getConfiguration(), "test")
    }

    @AfterEach
    fun cleanup() {
        if (::applicationContext.isInitialized) {
            applicationContext.close()
        }
    }

    protected open fun getConfiguration(): Map<String, Any> {
        val broker = UUID.randomUUID().toString().replace("-", "")
        return mutableMapOf(
            "micronaut.jms.activemq.classic.connectionString" to "vm://$broker?broker.persistent=false",
            "micronaut.jms.activemq.classic.enabled" to true,
            "micronaut.jms.activemq.classic.username" to "activemq",
            "micronaut.jms.activemq.classic.password" to "activemq",
            "spec.name" to javaClass.simpleName
        )
    }

    protected fun getConnectionPool(): JMSConnectionPool {
        return applicationContext.getBean(JMSConnectionPool::class.java, Qualifiers.byName(CONNECTION_FACTORY_BEAN_NAME))
    }
}
